### ISSUE TITLE
MultiplayerSynchronizers are not able to be moved in your project files

### ISSUE THREAD

**Issue by hunterkepley** [godotengine/godot#76411]
### Godot version

4.1.dev1

### System information

MacOS, Windows, Linux -- Mac->M2 MacBook 13" [Vulkan] Windows->1080Ti and ryzen 5 1600x [GLES3] Linux->1080Ti and ryzen 5 1600x [Vulkan]

### Issue description

When moving a scene file [`.tscn`] into a new folder (even when using the built in move to functionality) causes errors to occur with scenes containing MultiplayerSynchronizers (from here on, `MS's`):

![](https://i.imgur.com/JjYaJzn.png)

If you reparent a node which is a scene containing an MS in an RPC func like so:

```
@rpc("any_peer", "call_local")
func update_objects_in_eat_queue(delta):
	for eat_queue_object in object_eat_queue:
		var object = get_node(eat_queue_object.object_path)
		if object != null:
			object.set_meta("slide_timer", object.get_meta("slide_timer") + 1.0 * delta)
			if object.get_meta("slide_timer") >= EAT_QUEUE_TIMER_MAX:
				object.set_owner($SlimeBody/Objects.get_tree().get_edited_scene_root())
				object.get_parent().remove_child(object)
				$SlimeBody/Objects.add_child(object)
				if objects_height_offset > 0:
					objects_height_offset -= 0.01
				object.global_position = $SlimeBody.global_position + eat_queue_object.offset
				object_eat_queue.erase(eat_queue_object)
			else:
				object.global_position = object.global_position.move_toward($SlimeBody.global_position + eat_queue_object.offset, OBJECT_SPEED*delta)
		else:
			object_eat_queue.erase(eat_queue_object)
```

If anyone could help, or get this bug fixed, it would help me out so much. Currently I am having to recreate MS's every time I move a scene file in my project... makes for a big waste of time :(

### Steps to reproduce

Move a scene file containing an MS using Godot's built in tool to a new folder

No reproduction project needed as I believe this bug is trivial:
1. create a scene
2. create a second scene
3. add an MS to the second scene
4. import scene 2 into scene 1
5. Use Godot's "move to" functionality on scene 2 into a new folder by right clicking a file
6. Run game and watch errors pile in.

### Minimal reproduction project

N/A

**Comment by hunterkepley** [comment]
You also can't queue free MS's inside scenes.

```
E 0:00:12:0497   get_node: Node not found: "Multiplayer/Level/World/WorldEnvironment/Players/233141643/Slime/SlimeBody/Objects/Nut31/MultiplayerSynchronizer" (relative to "/root").
  <C++ Error>    Method/function failed. Returning: nullptr
  <C++ Source>   scene/main/node.cpp:1384 @ get_node()
E 0:00:18:0420   process_simplify_path: Condition "node == nullptr" is true.
  <C++ Source>   modules/multiplayer/scene_cache_interface.cpp:77 @ process_simplify_path()

```

Over and over after deleting one.

**Comment by hunterkepley** [comment]
This bug is either fixed or much better in Godot 4.1.dev1, closing
